controller/konnect/reconciler_konnectapiauth.go

```diff
@@ -0,0 +1,159 @@
+package konnect
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	sdkkonnectgoops "github.com/Kong/sdk-konnect-go/models/operations"
+	konnectv1alpha1 "github.com/kong/kubernetes-configuration/api/konnect/v1alpha1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kong/gateway-operator/controller/pkg/log"
+	k8sutils "github.com/kong/gateway-operator/pkg/utils/kubernetes"
+)
+
+// KonnectAPIAuthConfigurationReconciler reconciles a KonnectAPIAuthConfiguration object.
+type KonnectAPIAuthConfigurationReconciler struct {
```

Comment:
Would it stutter to use `konnect.Konnect***` as some linters may suggest?

Reply:
On one hand yes, that's true. On the other hand though we'd ideally want this to have a "konnect" prefix the same way as the CRDs (e.g. added in https://github.com/Kong/kubernetes-configuration/pull/12). 

So just for consistency I'd keep this as is even though it stutters.

---

Listing those CRDs is then unambiguous using e.g. `kubectl` using short resource names: so we end up with 

```
kubectl get konnectapiauthconfigurations
# equivalent of
kubectl get konnectapiauthconfigurations.konnect.konghq.com
```

and not

```
kubectl get apiauthconfigurations
# equivalent of
kubectl get apiauthconfigurations.konnect.konghq.com
```

---

For the full context:

I believe this can be overriden using kubebuilder code markers like so:

```
// +kubebuilder:resource:scope=Namespaced,path=konnectapiauthconfigurations,singular=konnectapiauthconfiguration
```